### PR TITLE
[JENKINS-41502] Select provider using a paramater

### PR DESCRIPTION
**Pull request by imeredith**
Lets the provider be selected with a http param. Falls back to existing methods if it cant find a matching one

part of : https://issues.jenkins-ci.org/browse/JENKINS-41502

### CODE REVIEW

**Comment by michaelneale** [comment]
@i386 do you mind reviewing this change as I don’t know things well enough here.

File: src/main/java/org/jenkinsci/plugins/displayurlapi/actions/AbstractDisplayAction.java [hunk 1]
```diff
@@ -34,13 +36,36 @@ public String getUrlName() {
...
         return null;
     }
 
     protected abstract String getRedirectURL(DisplayURLProvider provider);
 
+    DisplayURLProvider lookupProvider(StaplerRequest req) {
+        final String providerName = req.getParameter("provider");
+        if(providerName != null && !providerName.isEmpty()) {
+            Iterable<DisplayURLProvider> providers = DisplayURLProvider.all();
+            Iterable<DisplayURLProvider> filtered = Iterables.filter(providers, new com.google.common.base.Predicate<DisplayURLProvider>() {
```
Comment (i386) [inline src/main/java/org/jenkinsci/plugins/displayurlapi/actions/AbstractDisplayAction.java]: nit: import Predicate

**Comment by michaelneale** [comment]
yeah nice tests 👏

File: src/main/java/org/jenkinsci/plugins/displayurlapi/actions/AbstractDisplayAction.java (hunk 1 above)
Comment (imeredith) [reply src/main/java/org/jenkinsci/plugins/displayurlapi/actions/AbstractDisplayAction.java]: oops, i had imported java predicate and then intellij did the above

**Pull request merged**